chains/ethereum/connection.go

```diff
@@ -153,3 +154,21 @@ func (c *Connection) ensureHasBytecode(addr ethcommon.Address) error {
 	}
 	return nil
 }
+
+// waitForBlock will poll for the block number until the current block is equal or greater than
+func (c *Connection) waitForBlock(block *big.Int) error {
```

Comment:
Should we set a timeout for the wait for block? Ie if something happens where we don't see a block for x amount of BlockRetryIntervals, would it make sense to error out or at least put out a warning of unusual block times?

Reply:
I think if we don't see new blocks and there is no error returned, the node must be stalled. In this case it would be better to wait for the node to catch up, if it doesn't a manual restart is required and this will be killed. How does that sit with you?

Reply:
I think if no error is returned and the block never increments there are bigger issues than this not returning. if this happens the system will never proceed as no blocks are being produced to initiate transfers or finalize proposals. 

We definitely need to consider error cases a bit more, this is a great point to bring up. In the ideal case, if we loose connection, this will wait until we reconnect and then continue.